kratos/utilities/properties_configuration.h

```diff
@@ -0,0 +1,356 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+
+#if !defined(KRATOS_PROPERTIES_CONFIGURATION_H)
+#define  KRATOS_PROPERTIES_CONFIGURATION_H
+
+// System includes
+#include <map>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "containers/data_value_container.h"
+#include "geometries/geometry.h"
+#include "includes/node.h"
+#include "includes/process_info.h"
+#include "includes/properties.h"
+
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+class PropertiesConfiguration
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of PropertiesConfiguration
+    KRATOS_CLASS_POINTER_DEFINITION(PropertiesConfiguration);
+
+    using PropertiesType = Properties;
+    using NodeType = Node<3>;
+    using GeometryType = Geometry<NodeType>;
+
+    using IndexType = std::size_t;
+    using SizeType = std::size_t;
+
+    using DoubleVariableType = Variable<double>;
+    using ComponentVariableType = VariableComponent< VectorComponentAdaptor<array_1d<double, 3> > >;
+    using Array3VariableType = Variable<array_1d<double, 3>>;
+
+    struct ConfigurationParameters
+    {
+    public:
+
+        ConfigurationParameters(const PropertiesType& rProps,
+                                const GeometryType& rGeom,
+                                const DataValueContainer& rData) :
+                                mpProperties(&rProps),
+                                mpGeometry(&rGeom),
+                                mpData(&rData)
+                                {}
+
+        const PropertiesType& GetProperties() const
+        {
+            KRATOS_DEBUG_ERROR_IF_NOT(mpProperties) << "Properties is nullptr!" << std::endl;
+            return *mpProperties;
+        }
+
+        const GeometryType& GetGeometry() const
+        {
+            KRATOS_DEBUG_ERROR_IF_NOT(mpGeometry) << "Geometry is nullptr!" << std::endl;
+            return *mpGeometry;
+        }
+
+        const DataValueContainer& GetDataValueContainer() const
+        {
+            KRATOS_DEBUG_ERROR_IF_NOT(mpData) << "DataValueContainer is nullptr!" << std::endl;
+            return *mpData;
+        }
+
+    private:
+        const PropertiesType*     mpProperties;
+        const GeometryType*       mpGeometry;
+        const DataValueContainer* mpData;
```

Comment:
Does mpData Need to be a member? GetValue Forwards to properties but mpData also has a get value which makes it a bit confusing and I don´t see where mpData is needed.